src/zarr/abc/numcodec.py

```diff
@@ -0,0 +1,54 @@
+from typing import Any, Self, TypeGuard
+
+from typing_extensions import Protocol
+
+
+class Numcodec(Protocol):
+    """
+    A protocol that models the ``numcodecs.abc.Codec`` interface.
+    """
+
+    codec_id: str
+
+    def encode(self, buf: Any) -> Any: ...
```

Comment:
Maybe it'd be helpful to at least have a alias for Any to make it more clear that this will be eventually restricted to Buffer-Like objects?